src/components/rating/Rating.js

```diff
@@ -0,0 +1,63 @@
+import { useState } from "react";
+import RatingComponent from "./RatingComponent";
+import styles from "@/styles/reviews.module.scss";
+
+const Rating = (props) => {
+  const [hovering, setHovering] = useState(false);
+  const [hoveringIndex, setHoveringIndex] = useState(0);
+
+  let rating = [];
+  for (let i = 1; i < 6; i++) {
+    if (hovering) {
+      if (i <= hoveringIndex) {
+        rating.push(
+          <RatingComponent
+            id={i}
+            setRating={props.setRating}
+            setHovering={setHovering}
+            setHoveringIndex={setHoveringIndex}
```

Comment:
It's generally a risky thing to pass in the state setters from one component (`Rating`)  to another (`RatingComponent`) because it means the child now can directly control the parent, and the parent also knows about the child. This means that both components understand/interact directly with eachother which means poor separation of concerns.